tests/lib/alertmanager_api/test_silence_details.py

```diff
@@ -0,0 +1,213 @@
+from datetime import datetime
+import json
+import pytest
+from alertmanager_api.silence_details import SilenceDetails, SilenceDetailsHandler
+
+
+@pytest.mark.parametrize(
+    "attr, value, expected_message",
+    [
+        ("start_time_dt", "not_dt", "start_time_dt must be of type datetime."),
+        ("end_time_dt", "not_dt", "end_time_dt must be of type datetime."),
+    ],
+)
+def test_invalid_datetime_fields(attr, value, expected_message):
+    """Test that SilenceDetails raises a TypeError for invalid datetime fields."""
+    kwargs = {
+        "instance_name": "name",
+        "start_time_dt": datetime(2025, 1, 1, 10, 0, 0),
+        "end_time_dt": datetime(2025, 1, 1, 10, 0, 0),
+        "author": "author",
+        "comment": "comment",
+    }
+    kwargs[attr] = value  # Set the invalid attribute
+
+    with pytest.raises(TypeError, match=expected_message):
+        SilenceDetails(**kwargs)
+
+
+@pytest.mark.parametrize(
+    "status, expected_active, expected_pending, expected_expired, expected_valid",
+    [
+        ("active", True, False, False, True),  # "active" should be valid
+        ("pending", False, True, False, True),  # "pending" should be valid
+        ("expired", False, False, True, False),  # "expired" is not valid
+    ],
+)
+def test_status_properties(
+    status, expected_active, expected_pending, expected_expired, expected_valid
+):
+    """Test that is_active, is_pending, is_expired, and is_valid return correct values for each status."""
+    instance = SilenceDetails(status=status)
+
+    assert instance.is_active == expected_active
+    assert instance.is_pending == expected_pending
+    assert instance.has_expired == expected_expired
+    assert instance.is_valid == expected_valid
+
+
+# ==============================================================================
+#   test class SilenceDetailsHandler
+# ==============================================================================
+
+
+def test_add_from_json_with_string():
+    """Test that add_from_json correctly parses a JSON string into a list of dictionaries."""
+
+    json_input = '[{"id": "123",\
+"startsAt": "2025-01-16T10:50:00.000Z",\
+"endsAt": "2025-01-16T10:51:00.000Z",\
+"createdBy": "admin",\
+"comment": "Testing.",\
+"status": {"state": "active"},\
+"matchers": [{"name": "instance", "value": "hv123.matrix.net"}]}]'
```

Comment:
We can improve coverage by adding this:
```suggestion
"matchers": [{"name": "env", "value": "prod"}, {"name": "instance", "value": "hv123.matrix.net"}]}]'
```